x-pack/plugins/observability_solution/inventory/public/pages/inventory_page/index.tsx

```diff
@@ -4,105 +4,73 @@
-            kuery,
-          },
-        },
-        signal,
-      });
-    },
-    [entityTypes, inventoryAPIClient, kuery, sortDirection, sortField]
+  const { view } = query;
+  const [inventoryState, setInventoryState] = useState<InventoryState | undefined>(() =>
+    view ? (decode(view) as InventoryState) : undefined
   );
 
-  useEffectOnce(() => {
-    const searchBarContentSubscription = searchBarContentSubject$.subscribe(
-      ({ refresh: isRefresh, ...queryParams }) => {
-        if (isRefresh) {
-          refresh();
-        } else {
-          inventoryRoute.push('/', {
-            path: {},
-            query: { ...query, ...queryParams },
-          });
-        }
-      }
-    );
-    return () => {
-      searchBarContentSubscription.unsubscribe();
-    };
-  });
+  useEffect(() => {
+    if (inventoryState) {
+      const encodedView = encode(inventoryState);
 
-  function handlePageChange(nextPage: number) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: { ...query, pageIndex: nextPage },
-    });
-  }
+      if (encodedView !== view) {
+        inventoryRoute.replace('/', {
+          path: {},
+          query: {
+            ...query,
+            view: encodedView,
+          },
+        });
+      }
+    }
+    // Only run the effect on a change for `inventoryState`. Other deps cause this to refire too many times.
+    // Probably a better way of doing this.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inventoryState]);
 
-  function handleSortChange(sorting: EuiDataGridSorting['columns'][0]) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: {
-        ...query,
-        sortField: sorting.id as EntityColumnIds,
-        sortDirection: sorting.direction,
-      },
-    });
-  }
+  const setGrouping = useCallback(
+    (kind: string) => setInventoryState((state) => ({ ...state, grouping: kind })),
+    []
+  );
+  const setPagination = useCallback(
+    (group: string, nextPage: number) =>
+      setInventoryState((state) => ({
+        ...state,
+        pagination: { ...state?.pagination, [group]: nextPage },
+      })),
+    []
+  );
 
-  function handleTypeFilter(entityType: EntityType) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: {
-        ...query,
-        // Override the current entity types
-        entityTypes: [entityType],
-      },
-    });
-  }
+  // If state or grouping itself is `undefined`, fallback to the default view
+  const activeView = inventoryState?.grouping || DEFAULT_VIEW;
 
   return (
-    <EntitiesGrid
-      entities={value.entities}
-      loading={loading}
-      sortDirection={sortDirection}
-      sortField={sortField}
-      onChangePage={handlePageChange}
-      onChangeSort={handleSortChange}
-      pageIndex={pageIndex}
-      onFilterByType={handleTypeFilter}
-    />
+    <InventoryPageViewContextProvider
+      grouping={activeView}
```

Comment:
I think the naming here is a bit off.... By looking at this code for the first time I don't know what's the connection between `grouping` and `activeView`.